src/services/server/ServerService.ts

```diff
@@ -66,6 +105,34 @@ export class ServerService {
     return deferred.promise;
   }
 
+  getServerStats(server: IServerAddress): ng.IPromise<any> {
+    var deferred: ng.IDeferred<string[]> = this.$q.defer<string[]>();
+    var request: IDmrRequest = <IDmrRequest>{
+      address: [].concat("host", server.host, "server", server.name, "core-service", "platform-mbean"),
+      recursive: true,
+      "child-type": "type",
+      "include-runtime": true,
+      "recursive-depth": 2
+    };
+    this.dmrService.readChildResources(request).then(
+        (response) => deferred.resolve(response),
+        () => deferred.reject());
+    return deferred.promise;
```

Comment:
First call this.getServerStatus() and if not stopped try to retrieve the stats. Simply rejecting the deferred promise if the status is stopped should be sufficient IMO.